llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -5982,7 +5982,9 @@ SDValue DAGCombiner::hoistLogicOpWithSameOpcodeHands(SDNode *N) {
         LegalTypes && !TLI.isTypeDesirableForOp(LogicOpcode, XVT))
       return SDValue();
     // logic_op (hand_op X), (hand_op Y) --> hand_op (logic_op X, Y)
-    SDValue Logic = DAG.getNode(LogicOpcode, DL, XVT, X, Y);
+    SDNodeFlags LogicFlags;
+    LogicFlags.setDisjoint(N->getFlags().hasDisjoint());
```

Comment:
Just to make sure I understand this right.  We've got (or disjoint (ext a), (ext b)) and we're turning that into (ext (or disjoint a, b)).  Right?

For zext, this is fine.  For sext, this is fine.  For the in_reg variants, I am not sure if this is fine or not.  What if a and b had the same high bit set and we're doing a sext_in_reg which clears that bit?  I think that violates the disjoint and introduces UB which didn't previously exist right?

Reply:
Agh good catch, alive2 does indeed confirm this: https://alive2.llvm.org/ce/z/W9F5b8. Will fix